### PR TITLE
Refactor and add filters to members page

### LINKED ISSUE #3381: Add pagination to contributors & watchers list on the members page
https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=5015%3A57349

10 watchers/contributors per page

### PR DESCRIPTION
**Pull request by chinins**
## Description

This PR adds filters to the members page. It also refactors the code a bit to split and simplify it. I've also added fixed width to the MembersListItem so that watchers item list doesn't shrink when there are no contributors.

Please, note that the hover state of the `reset` button for the filters is not as in the specs. It was agreed with Arren that it will stay the same as all other buttons above.

Update figma specs
https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=7705%3A91695

Screenshot 2022-07-01 at 16 08 23

**New stuff** ✨

- add filters checkboxes and logic

**Changes** 🏗

- extract MemberControls to a separate component
- fix the MembersListItem width

resolves #3381

### CODE REVIEW

**Review by willm30 — CHANGES_REQUESTED** [review]
Nice work! 

I noticed a few things as I was going through:
1. When the team filter is applied, the [invite link should be hidden](https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=6736%3A94732)
invite-menu

2. There aren't the numbers next to each filter indicating the number of matches
no-numbd

3. The filters jump when nothing is selected
jumpy-filter

4. I think it would be good to include a simple message here when nothing is selected. Something like "No filters selected." would suffice.
no-members-msg

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
For me, that I have a large screen to test it, it doesn't jump like the gif that WIll uploaded, however, it does seem like the width doesn't  adapt to the container:
FireShot Capture 649 - Members - Colony - wonker - localhost

**Comment by chinins** [comment]
@willm30, @ArmandoGraterol, I've fixed the width adjustments. Please, let me know if it's working properly for you.

I've also removed the invite link when it's not a root domain.

As for the count in the filter checkbox labels, it's been decided  with Arren not to have it.

**Comment by chinins** [comment]
> @chinins Looks good! Pre-approving. Just flagging the following error that's appearing in the console when loading the members page:

It's not an error, it's a warning. I'm aware of it. It also happens in several places in the dapp so it's not a big deal.

File: src/modules/dashboard/components/Members/filterMembers.ts [hunk 1]
```diff
@@ -1,21 +1,64 @@
...
+      }
+
+      return !isWhitelisted;
+    });
+  }
+
+  /* All the filters together */
+  return data.filter(({ banned, isWhitelisted, profile, id }) => {
+    const textFilter =
+      profile?.username?.toLowerCase().includes(filterValue?.toLowerCase()) ||
```
Comment (danbr) [inline src/modules/dashboard/components/Members/filterMembers.ts:48]: This block is almost a duplicate of the above, just without `textFilter`. Could this be refactored?

**Review by danbr — APPROVED** [review]
Very good. Works for me. Approving but have a comment (in the code)....

**Review by arrenv — CHANGES_REQUESTED** [review]
Working well! I noticed the filter doesn't work to show only Verified and Banned.

download (32)

download (33)

download (34)

**Comment by ArmandoGraterol** [comment]
> @willm30, @ArmandoGraterol, I've fixed the width adjustments. Please, let me know if it's working properly for you.
 
It's looking good on my end now! Thank you

**Review by arrenv — APPROVED** [review]
It is working really well, nice one!

**Review by willm30 — CHANGES_REQUESTED** [review]
Filters looking great! Noticed a couple of things re. functionality: 

1. Not banned contributors are not showing up in "Not banned"
banned-members

2. Verifying someone overrides previous verification. To reproduce: 
* Verify somebody and refresh
* Verify somebody else and refresh
* The first person is no longer verified. 

And a tiny thing re container width: 

3. The container shrinks slightly when the filters return no entries
collapsing-title-members

**Review by danbr — COMMENTED** [review]
(UN)Fortunately, I dont have the issues that @willm30 has experienced.

For me, the 'Banned' dropdown is at the bottom of page, so clicking dont give any obvious reaction as the dropdown options are off page.
It could be worth considering scrolling the page down when clicking on the 'Banned' dropdown.

(Also not great to have the feedback link on top of the dropdown)

Screenshot 2022-07-06 at 21 40 17

**Review by ArmandoGraterol — COMMENTED** [review]
For me, the contributor appears when the `Not banned` filter is applied but the watcher is disappearing:
FireShot Capture 655 - Members - Colony - wonker - localhost

FireShot Capture 656 - Members - Colony - wonker - localhost

Other than that, I haven't noticed anything strange

**Comment by willm30** [comment]
I have re-tested this morning. I'm still noticing 2 & 3, so will create a separate issue for 2. 3 (container shrink) occurs when filtering to "Watchers". 

As for banned filter, I believe it's related to the verified address. I.e., for me, when the member is verified, they are not being included at all by the banned filter. Have reproduced again with a different example (**@d is the member to watch**): 

banned-behaving-off

**Comment by arrenv** [comment]
> As for banned filter, I believe it's related to the verified address. I.e., for me, when the member is verified, they are not being included at all by the banned filter. Have reproduced again with a different example (**@d is the member to watch**):

I tested again just now again as well, I am experiencing the same thing with the same combo.

Member type: All members
Verification type: Any
Banned status: Not banned

Member who is verified disappears, when they shouldn't.

**Comment by chinins** [comment]
@willm30, @ArmandoGraterol, I've fixed the banned bug.

@danbr, I've added the scroll.

@willm30, as discussed, nobody can reproduce the jumping screen. Please, raise an issue and work on it since you are the only one who experience this problem.

**Review by willm30 — APPROVED** [review]
I have created the following issues to follow up with things I noticed here:

1. [Members List width shrinks when filter watchers](https://github.com/JoinColony/colonyDapp/issues/3593)
(**Note, to reproduce you need to verify the contributor but not the watcher (see issue for set up)**)

2.  [Address book only verifying one member at a time](https://github.com/JoinColony/colonyDapp/issues/3588)

As such, approving as the banned filter is working as expected and the rest works great.

**Comment by chinins** [comment]
> I have created the following issues to follow up with things I noticed here:
> 
> 1. [Members List width shrinks when filter watchers](https://github.com/JoinColony/colonyDapp/issues/3593)
>    (**Note, to reproduce you need to verify the contributor but not the watcher (see issue for set up)**)
> 

With this setup, I can reproduce. So it's an issue up for grabs for anybody. Good catch about the difference.

**Review by ArmandoGraterol — APPROVED** [review]
Working great! Excellent job Olga :) Tested a lot of combinations and it's working as expected

**Review by danbr — APPROVED** [review]
The auto scrolling updates work well. It all LGTM.

**Pull request merged**